### PR TITLE
T-18127 staging to master: cache refresh refactor

### PR DESCRIPTION
**Pull request by commoddity**
this PR adds a requested refactor to move cache handler logic to inside the cache package.

### CODE REVIEW

**Review by pablocampogo — DISMISSED** [review]

File: cache/cache.go [hunk 1]
```diff
@@ -60,8 +59,174 @@ func NewCache(reader driver.Reader, logger *logrus.Logger) *Cache {
...
+		blockchainsMap[blockchain.ID] = blockchain
+	}
+
+	c.blockchains = blockchains
+	c.blockchainsMap = blockchainsMap
+
+	return nil
+}
+
+func (c *Cache) setApplications() error {
```
Comment (adshmh) [inline cache/cache.go:162]: Not a blocker, but moving this function (and a few others) around in the same code file has made the PR look much larger than it is. This kind of change should ideally be reserved for a separate PR (or at-least a separate commit) to make the PR review easier.

**Review by adshmh — DISMISSED** [review]
Thank you for the PR! Moving the logic to the cache package makes sense IMO.
LGTM.

**Review by adshmh — APPROVED** [review]
Thank you for adding the log entries for monitoring. LGTM

**Review by pablocampogo — APPROVED** [review]

**Review by ricantar — APPROVED** [review]
